### PR TITLE
Corgstation 4

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

The 4th iteration of a map I've been casually doing whenever I don't feel like coding is finally complete.
Corgstation is an entirely new and completely original station with a unique design (Whether or not this is a good thing is still to be found).
The map features absolutely beautiful lighting and introduces a new glowsticks spawner that will auto light glowsticks on the groups for giving a colourful and pretty glow to maints.
No one really cares about the text, let's get to the pretty pictures.

I'm quite happy how most of it turned out, brig physician is going to have a good time, although I think atmospherics is a little off.

The engine used is a standard supermatter with valves that can be opened and closed depending on whether or not you want to connect it to the TEG, which can also be connected to the cold loop or turbine. (It's pretty customisable, you can choose what you want connected by opening and closing valves.)
Robotics is extremely nice, very happy with how that turned out.
Cargo I think is quite nice too and I like security.
Genetics and medbay storage are pretty good.
Evac is beautiful and cozy.

Also the maintenance game is very strong for you assistants out there.

image

Also the prisons are super nice for your long term stays.

## Composite Map

(OUTDATED DUE TO UPDATES, SEE CHECKS FOR UP TO DATE IMAGE)

<a href="https://ibb.co/cJHnd5b">corgstation-full</a>

image

## Highlights

My hand-selected favourite parts of the map.

image

image

image

image

image

image

image

image

image

image

image

image

## Why It's Good For The Game

A pretty nice new map.

## Changelog
:cl:
add: Corgstation
add: Lit glowsticks spawner.
add: Corgstation code ownership
add: Corgstation Emergency Shuttle
add: Corgstation Arrivals Shuttle
fix: Fixes force_map not actually working
tweak: Allows shuttles to be created in area/science/shuttle
tweak: Pins some important vars on fastdmm2 for obj/machinery/camera
tweak: Allows anchored supermatter crystals to be launched via those chapel cannon thingys
/:cl:

### CODE REVIEW

**Comment by Crossedfall** [comment]
ooooooooo

**Comment by PowerfulBacon** [comment]
ah shit I failed the tests 😔

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -28,6 +28,7 @@ TGS3.json @crossedfall
 
 /code/game/machinery/shuttle @powerfulbacon
 /code/modules/shuttle @powerfulbacon
+_maps/map_files/CorgStation @powerfulbacon
```
Comment (PowerfulBacon) [inline .github/CODEOWNERS]:
```suggestion
/_maps/map_files/CorgStation @powerfulbacon
```

File: code/datums/map_config.dm [hunk 2]
```diff
@@ -54,7 +54,7 @@
 		log_world("Could not open map_config: [filename]")
 		return
 
-	json = rustg_file_read(json)
+	json = file2text(json)
```
Comment (PowerfulBacon) [inline code/datums/map_config.dm]:
```suggestion
	json = rustg_file_read(json)
```

**Comment by ghost** [comment]
What happened to Corgstation 1, 2 and 3?

**Comment by PowerfulBacon** [comment]
> 
> 
> What happened to Corgstation 1, 2 and 3?

1 I made entirely without saving (actually I'm pretty sure I saved it a few times and backed it up to git but I still managed to lose it), 2 and 3 were shit and I restarted early on.

**Comment by PowerfulBacon** [comment]
Horray it passes

**Comment by PowerfulBacon** [comment]
oh fuck I forgot the labour shuttle shit

**Comment by PowerfulBacon** [comment]
Gulag shuttle will go on the inside

**Comment by ike709** [comment]
Did you include all of the miscellaneous spawners in space/maint and did you proper use the nearstation area? Those are usually the biggest gotchas for wholly original maps.

**Comment by ike709** [comment]
Map looks pretty good though.

**Comment by PowerfulBacon** [comment]
> 
> 
> Did you include all of the miscellaneous spawners in space/maint and did you proper use the nearstation area? Those are usually the biggest gotchas for wholly original maps.

Station uses nearstation, and yea I forgot the carp spawners for most of the station

**Comment by ghost** [comment]
image
wtf bacon, why can you stare at people showering?

**Comment by LastCrusader105** [comment]
EndoStation best station

legit doe wheres the full station image?

**Comment by St0rmC4st3r** [comment]
Is it intended to be a lowpop map?

**Comment by Crossedfall** [comment]
> 
> 
> EndoStation best station
> 
> legit doe wheres the full station image?

You get them from the checks.

https://github.com/BeeStation/BeeStation-Hornet/pull/3085/checks?check_run_id=1501545349

**Review by park66665 — DISMISSED** [review]
Screenshot 2020-12-05 140740

Found at X 158 Y 148. It shouldn't be a manifold. Weird pipe connections are not just a map bug; preliminary results show that such mis-connections might sabotage pipeline code functions occasionally.

This map is great anyway, don't worry much about it. Box probably has multiple of them and it doesn't break BYOND.

**Comment by LastCrusader105** [comment]
> This map is great anyway, don't worry much about it. Box probably has multiple of them and it doesn't break BYOND.

Box won’t when they merge my PR, together we will bring a new standard to mapping!

**Comment by PowerfulBacon** [comment]
> image
> wtf bacon, why can you stare at people showering?

There are shutters haha

**Comment by PowerfulBacon** [comment]
> EndoStation best station
> 
> legit doe wheres the full station image?

Look at composite image in the pr description for a full render of the map with lighting

**Comment by PowerfulBacon** [comment]
Needs a delivery beacon network, that pipe fixed and generic event spawned.

**Comment by PowerfulBacon** [comment]
Ready

**Comment by ZombiesVsPlants** [comment]
Looking at the Ai sat as an ai main I can see 4 problems so far
1. Where are those mounted flashes? Without them, there is nothing stopping an unsynced borg from just waltzing in and bludgeoning your core to death since the "Target cyborgs" option for the turrets doesn't actually work.
2. Where are the shutters? On Metastation you have 2 buttons to put shutters
Here:
image
and the next best place I'd assume here:
image
On the windows and door
The first shutter is required otherwise people can just break in through the top and shoot you with a laser and kill you outright.
The 2nd shutter is not AS necessary but it was a nice layer of defense on Metastation and would be required to defend your apc if you added point 3.
3. The APC at the back, while very defensible, only encourages people to break in through the top and block using unsecured girders or closets, and breaking the apc is an instant win. I would move it to the front to encourage people to come through the front door and force people going in the backway to fight the ai core head-on. (Something the ai can defend against with pingsky, flashes, turrets, and soap)
4. I can see VERY few cameras in the ai sat, 1 camera in the core I doubt would cover the entire core room and nothing is sadder than having no vision in your very own core room.
Yeah comparing this and metastation I can only see 1 camera in the carpstation core and 4 cameras in the metastatation core (Which is far smaller)

I would also worry if the single SMES is enough to power the many more turrets, Telecomms, and ai upload for 2 hours, though beestation might have far faster rounds than AUstation.

Other than that it looks perfect and I'm looking forward to playing it.

**Comment by ZombiesVsPlants** [comment]
OH also 
image
how is the captain meant to turn off the turrets before entering?

**Comment by ZombiesVsPlants** [comment]
also, I would appreciate a few more holopads for the ai core if at all possible.

**Comment by ZombiesVsPlants** [comment]
oh and no cyborg recharger as far as I can see.

**Comment by PowerfulBacon** [comment]
> 
> 
> OH also
> image
> how is the captain meant to turn off the turrets before entering?

The turret controller for that room is just before the bridge thing

**Comment by PowerfulBacon** [comment]
Valid points, I plan to rework medical and AI core today

**Comment by PowerfulBacon** [comment]
image

**Comment by PowerfulBacon** [comment]
After some work changes have been made:

Medical:
 - Old cryo room replaced with morgue.
 - Cryo pods moved next to sleeper room.
 - Chemistry has been reorganised and a chem wardrobe added.
 - Added shutters to CMO's office.

Permabrig:
 - Added in exterior protection.

AI Core:
 - Relocated turrets
 - Added shutters to windows
 - Added mounted flashers
 - Added cyborg charging station

Robotics:
 - Added cyborg charging station

Chapel:
 - Removed maintenance room
 - Added crematorium

Head of Personnel's office:
 - Reduced in size and extra space added to maints
 - Added 2 posters

Security: Added the prisoner transfer center

Virology:
 - Added space reinforcements
 - Linked disposals up
 - Added 2 space suits

Also:
 - Added solars

**Comment by ZombiesVsPlants** [comment]
A camera next to the primary core would be greatly appreciated.

**Comment by Penwin0** [comment]
I am unable to find any major gripes with the map, its good

**Comment by ghost** [comment]
I don't like how there are two spare cartons of Synthflesh in genetics, cloning should be discouraged not encouraged.

**Comment by PowerfulBacon** [comment]
> 
> 
> I don't like how there are two spare cartons of Synthflesh in genetics, cloning should be discouraged not encouraged.

Cloning and surgery should be encouraged and used for their intention: Surgery for reviving simple damages / dead bodies, cloning for the harder to heal stuff like husks or brains. Pretty sure all other maps have synthflesh around

**Comment by ghost** [comment]
> Pretty sure all other maps have synthflesh around

No, they don't, all other maps only have one Synthflesh carton in the CMO's locker, that's it, and it's for a good reason.

**Comment by PowerfulBacon** [comment]
Added:
 - Gateway
 - Escape Pods

File: interface/interface.dm [hunk 3]
```diff
@@ -264,6 +264,7 @@ Any-Mode: (hotkey doesn't need to be on)
 		if("MetaStation")			map_in_url = "meta"
 		if("Kilo Station")          map_in_url = "kilo"
 		if("PubbyStation")          map_in_url = "pubby"
+		if("CorgStation")			map_in_url = "corg"
```
Comment (PowerfulBacon) [inline interface/interface.dm:267]: Will need website update

**Comment by PowerfulBacon** [comment]
There is no PDA server

**Comment by PowerfulBacon** [comment]
Solars need solars

**Comment by PowerfulBacon** [comment]
no sugar or flower

**Comment by PowerfulBacon** [comment]
No genetics computer in botany

**Comment by PowerfulBacon** [comment]
No science servers

**Comment by PowerfulBacon** [comment]
Botany doesnt get their disk stuff, no grinder. no tables. No kitchen access. no chem machine

**Comment by PowerfulBacon** [comment]
toxins chamber doors dont open

**Comment by PowerfulBacon** [comment]
shaft miners, have no cargo access. and therefore cant go to mining station

**Comment by PowerfulBacon** [comment]
one of the xenobio slime cells is missing walls and its just spaced

**Comment by PowerfulBacon** [comment]
I dont think sec starts with a techfab still

**Comment by PowerfulBacon** [comment]
Antique gun

**Comment by PowerfulBacon** [comment]
uh botanys chem machine seeems to be broken?

**Comment by PowerfulBacon** [comment]
no tracking implants

**Comment by PowerfulBacon** [comment]
north-east maint still lacks an airlock

**Comment by PowerfulBacon** [comment]
i cant open the sec checkpoint near the shuttle as a secoff

**Comment by PowerfulBacon** [comment]
Ok who the fuck decided to make the brig entry all access.

**Comment by PowerfulBacon** [comment]
Reee fastdmm is really fucking annoying and keeps resettings variables of buttons for no reason

**Comment by PowerfulBacon** [comment]
add coffee machine

**Comment by Effemu** [comment]
okay so 2 things

- med needs prosthetics box
- surgery room doesnt have circular saw and retractor

there was something else about med too but i forgot what it was

**Comment by PowerfulBacon** [comment]
assistant has access to clown's room

**Comment by Sinestia** [comment]
cat ears in bomb test site :'(

**Comment by doofus9341** [comment]
- security camera in engineering by the emitters should be moved since the emitters destroy it when you turn them on
- some rapid cable layers in engineering would be nice

**Review by utilbird — DISMISSED** [review]

**Comment by PowerfulBacon** [comment]
ill do some more testmerging soon (tm)

**Comment by PowerfulBacon** [comment]
Needs more holopads

**Comment by KennyBadrige** [comment]
hey, after playing cargo for a bit, i compiled my gripes into an imgur album: https://imgur.com/a/Bn8TqfR

**Comment by Cenrus** [comment]
Dorms doesn't have a ClothesMate vendor, unless I am blind

**Comment by Cenrus** [comment]
Nightmare was unable to spawn on the map, does it have xeno spawn points?

**Comment by PowerfulBacon** [comment]
> 
> 
> Nightmare was unable to spawn on the map, does it have xeno spawn points?

Yes, many.
Note sure whats up with nightmare they have been failing spawn checks on maps like meta recently

**Comment by PowerfulBacon** [comment]
redundant buttons in the old checkpoint

**Comment by PowerfulBacon** [comment]
No console to pay off your fines

**Comment by PowerfulBacon** [comment]
bridge grilles need shocking

**Comment by PowerfulBacon** [comment]
no virus food

**Comment by 4rchibald** [comment]
I just had my first experience with a map and I enjoyed it.
I really like the fact that we're finally breaking from Box and Meta designs and as engineer I especially enjoyed engineering department. Whole map has a modern vibe with conveyor belts used to transport food from kitchen too bar and purposefully designed room for shuttle construction and plumbing. 
Is that map inspired by something? I had a feeling of deja vu when I saw location of gateway, it took me back to my first days in SS13

Smaller issues I noticed:
-It seems that SM engine shutters would block emitter beam when lowered so I would remove middle shutter
-CE office could use some more windows since you can't peak inside it at all from engineering foyer
-Atmos desk seems to be a missing feature 
-Something that bothered me while checking map as ghost...shouldn't that curved railings at chemistry be placed the opposite way?

Bigger issues and suggestions that probably won't be ever changed since they are already too far gone and incorporated deeply into stations layout:
-Wasn't there a policy that we want distro and waste pipes on the same layers? Either I remember it completely wrong or this map goes against that design. I'm pretty sure Box was getting a lot of heat for its pipe layers and APC in maints
-As one of the few people who actually play with tcomms (fix them as engineer and mess with them as traitor) I am completely against putting them on AI sat. That really makes them closely guarded by AI who bolts them down and frequently check for its own safety. That makes it too hard and annoying to both fix and sabotage so I would suggest moving them inside the station
-Same goes for upload and vault
-I was working on medbay revamp for Box few months ago before I got a break from SS13. Of all the changes made I actually liked connection of chemlab, plumbing room and med storage. Felt it really worked well for chemistry-medical cooperation since it all connects at one spot. As doctor you don't need to go to storage to check for meds first and then look for chemist if meds are not there, chemists have easy way of delivering meds as well. Maybe if you will be messing with medbay layout at any time it would be worth trying fitting something similar in

95491561-0d702900-099a-11eb-8245-dca5149009c1
-

**Comment by heepox** [comment]
sec/cargo maint is a bit too isolated from main hall.

**Comment by PowerfulBacon** [comment]
> -Wasn't there a policy that we want distro and waste pipes on the same layers? Either I remember it completely wrong or this map goes against that design. I'm pretty sure Box was getting a lot of heat for its pipe layers and APC in maints

Pretty sure the policy was to have distro on layer 1 and the scrubbers on layer 3 and some older maps were updated to have this, although some maps use layer 2 for distro instead of later 1.

**Comment by MNarath1** [comment]
Atmos needs radsuits and plasma tanks in the tank storage no real reason why there should be non in them and radsuit is really needed especially for fusion/turbin also is it intentional that there is no access for atmos tech to the TEG?

**Comment by MNarath1** [comment]
And the scrubbers inside the turbine should have their own area/air alarm to prevent contamination thanks to someone setting air alarm in atmos to contaminated

**Comment by KennyBadrige** [comment]
SEC corgstation feedback
-no sec outposts anywhere
-brig literally has no sec consoles
-officers get the short end of the stick in terms of rooms, gear room thats it
-gulag shuttle takes up way too much space in relation to its utility
-gulag prisoners after collecting points come out infront of perma and the hos office and need to be escorted out of the brig. Otherwise leaving them with their items to break open perma.
-brig medical is way too huge.
-armory is only accesible via warden or hos bolting open 3 doors and a 1x1 corridor.
-airlocks right side of perma dont have firealarm doors.
-whole SE of the station serves no purpose but maint loot.

Honestly, i think it would be preferable to completely re-do the brig, in its current state, its infuriating to play on.

**Comment by FortSumter** [comment]
Did you add a dedicated custom shuttle construction yard? Would be nice if the custom station you built would give players the option to experience your other major contribution... :)

**Comment by PowerfulBacon** [comment]
> 
> 
> Did you add a dedicated custom shuttle construction yard? Would be nice if the custom station you built would give players the option to experience your other major contribution... :)

yes

**Comment by Sinestia** [comment]
:(

**Comment by 4rchibald** [comment]
@PowerfulBacon 

I'm really pleased to see Corg getting closer and closer to its final, finished state.
I'm also grateful that some of my suggestions were taken under consideration and you moved tcomms to station (I honestly doubt anyone else shared any kind of opinion about tcomms), that's a much better location for both crew and traitors to access them and I really like how they look.

Don't think anyone actually expressed appreciation for request consoles in almost every room, I hope it will make people use them more often.


I didn't have time to really participate in round that Corg was on but still found few small issues

First of all, why was conveyor system in kitchen removed? I actually really liked it. Corg station finally breaks away from standards set by Box and Meta, its not square but has all sorts of curves and bends , its not grey but colourful and its not static but interactive. Conveyor system to deliver food was actually a pleasing solution that both gave station some motion and wasn't some scripted gimmick, instead was something that players could replicate from tools and objects they are given, which I'm a big fan of. I'm curious about reasoning, was that players feedback or it didn't work as you intended?
That said, if you are really set on removing it you forgot about lever

lever

When I was space walking to wire solars I was surprised by turrets shooting at me, there should be some kind of barrier to either block turrets line of sight or simply just keep players far enough so they are out of range. Even from just passing by they managed to make me fall for a moment, I can imagine multiple situations in which it would be not only inconvenient but also deadly

turrets

This one is not really an issue, not sure how to even categorize it. There's just something odd about walls just connecting by corners, there should either be an opening that players can fit into (move that maint room a bit to right or just make it smaller by either one row or column of tiles) or it should be blocked by proper L shaped walls, in a way that telegraphs there is no passage, that its not just some artificial decision by someone who didn't want people to get in there for no real reason and just barely covered the spot

nopassage

Also weird cam placing outside perma and poster under airlock in maints east from engineering (clicked on it by mistake earlier so its ripped on my screenshot)

cams

poster


From just glancing at it there were some weird empty places in southern maints, not sure if random rooms didn't work there or they are just not filled yet (which is not surprising considering you are still moving things around so final maint shape may change)

**Comment by Cenrus** [comment]
image

Spacecurity?

File: code/controllers/subsystem/mapping.dm [hunk 4]
```diff
@@ -1,3 +1,5 @@
+//#define FORCE_MAP "_maps/corgstation.json"
+
```
Comment (ike709) [inline code/controllers/subsystem/mapping.dm]:
```suggestion
```

**Review by ike709 — DISMISSED** [review]

**Comment by PowerfulBacon** [comment]
the force maps needs to be there but should be box

**Comment by ike709** [comment]
oops please readd that

File: code/controllers/subsystem/mapping.dm [hunk 5]
```diff
@@ -1,3 +1,5 @@
+//#define FORCE_MAP "_maps/corgstation.json"
```
Comment (PowerfulBacon) [inline code/controllers/subsystem/mapping.dm]:
```suggestion
//#define FORCE_MAP "_maps/boxstation.json"
```

**Review by ike709 — APPROVED** [review]

**Review by Crossedfall — APPROVED** [review]

**Pull request merged**